youtube_dl/extractor/brainpop.py

```diff
@@ -2,42 +2,44 @@
 from __future__ import unicode_literals
 
 from .common import InfoExtractor
+from ..utils import (
+    remove_end
+)
 
 
 class BrainPOPIE(InfoExtractor):
-    _VALID_URL = r'https?:\/\/(?:(.+)\.)?brainpop\.com\/(?P<id>[^\r\n]+)'
+    _VALID_URL = r'https?:\/\/(?:(.+)\.)?brainpop\.com\/[^/]+/[^/]+/(?P<id>[^/?#&]+)'
     _TEST = {
         'url': 'https://www.brainpop.com/english/freemovies/williamshakespeare/',
         'md5': '676d936271b628dc05e4cec377751919',
         'info_dict': {
-            'id': 'english/freemovies/williamshakespeare/',
+            'id': '3026',
+            'display_id': 'williamshakespeare',
             'ext': 'mp4',
-            'title': 'William Shakespeare - BrainPOP',
+            'title': 'William Shakespeare',
             'thumbnail': 're:^https?://.*\.png$',
             'description': 'He could do comedies, tragedies, histories and poetry.  Learn about the greatest playwright in the history of the English language!',
         }
     }
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
-        webpage = self._download_webpage(url, video_id)
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
 
-        self.report_extraction(video_id)
+        content = self._parse_json(self._html_search_regex(r'var content = ([^;]*)', webpage, 'content'), display_id)
```

Comment:
Just get `['category']['unit']['topic']` right away, so you don't need to specify it every time